src/main/java/pl/zz/sharefood/service/FoodGetService.java

```diff
@@ -0,0 +1,46 @@
+package pl.zz.sharefood.service;
+
+import org.springframework.beans.BeanUtils;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import pl.zz.sharefood.domain.Food;
+import pl.zz.sharefood.domain.dto.FoodDto;
+import pl.zz.sharefood.payload.PagedResponse;
+import pl.zz.sharefood.repository.FoodRepository;
+import pl.zz.sharefood.util.ValidatePageUtil;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+public class FoodGetService {
+
+    private final FoodRepository foodRepository;
+
+    FoodGetService(FoodRepository foodRepository){
+        this.foodRepository = foodRepository;
+    }
+
+    public PagedResponse<FoodDto> getFood(List<String> foodNames, int page, int size){
```

Comment:
tutaj polecam nazwać funkcję execute(...). Wywołując ją będziesz miał foodListService.execute(...). To połączenie dobrze mówi co robi dana funkcja